### PR TITLE
mqsink(ticdc): add kafka client interface

### LINKED ISSUE #8051: add kafka client interface
add kafka client interface to hide sarama

### PR DESCRIPTION
**Pull request by sdojjy**
### What problem does this PR solve?

Issue Number: close #8051

### What is changed and how it works?
add a kafka interface and wrap the sarama implementation, this is the first PR of replacing sarama with kafka-go,  I will adjust the  producer interface in another PR

#### Tests 

 - Unit test
 - Integration test

### Release note 

```release-note
 `None`.
```

### CODE REVIEW

**Review by 3AceShowHand — APPROVED** [review]

File: pkg/sink/kafka/client.go [hunk 1]
```diff
@@ -21,4 +26,178 @@ type Client interface {
...
+		key []byte, value []byte) error
+
+	// Close shuts down the producer; you must call this function before a producer
+	// object passes out of scope, as it may otherwise leak memory.
+	// You must call this before calling Close on the underlying client.
+	Close() error
+}
+
+// AsyncProducer is the kafka async producer
+type AsyncProducer interface {
```
Comment (nongfushanquan) [inline pkg/sink/kafka/client.go:58]: is this interface only for sarama?

File: pkg/sink/kafka/client.go [hunk 2]
```diff
@@ -21,4 +26,178 @@ type Client interface {
...
+
+func (c *saramaKafkaClient) MetricRegistry() metrics.Registry {
+	return c.client.Config().MetricRegistry
+}
+
+func (c *saramaKafkaClient) Close() error {
+	return c.client.Close()
+}
+
+type syncProducer struct {
```
Comment (nongfushanquan) [inline pkg/sink/kafka/client.go:127]: SaramaSyncProuducer?

File: pkg/sink/kafka/client.go (hunk 1 above)
Comment (sdojjy) [reply pkg/sink/kafka/client.go:58]: for sarama and kafka-go, I will refine it in next PR

File: pkg/sink/kafka/client.go (hunk 2 above)
Comment (sdojjy) [reply pkg/sink/kafka/client.go:127]: fixed

**Review by nongfushanquan — APPROVED** [review]

**Review by asddongmen — APPROVED** [review]

**Pull request merged**